cloudscape-design/components
# fix: Add aria live announcer to code editor for errors and warnings

### Description

This PR adds an aria-live region to the code editor so that updates to the error and warning counts are announced to assistive technology.

The live region receives the same text that is being used to label the error and warning buttons.

Note: aria-live="assertive" has been used to ensure that the "polite" announcement of the characters typed does not cancel out the warning and to allow assistive technology to treat the announcement with importance

Resolves AWSUI-19219

### How has this been tested?

Verified functionality in NVDA, JAWS, and VoiceOver

To test:
* use the Code Editor "simple" page and cause an error to fire (for example add a "!" to the end of a line).
* observe output from screen readers when the error fires.


<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
YES
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
